src/wfs/src/test/java/org/geoserver/wfs/BatchManagerTest.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * (c) 2022 Open Source Geospatial Foundation - all rights reserved This code is licensed under the
+ * GPL 2.0 license, available at the root application directory.
+ */
+package org.geoserver.wfs;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.same;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.xml.namespace.QName;
+import net.opengis.wfs20.DeleteType;
+import net.opengis.wfs20.InsertType;
+import net.opengis.wfs20.NativeType;
+import net.opengis.wfs20.ReplaceType;
+import net.opengis.wfs20.TransactionType;
+import net.opengis.wfs20.UpdateType;
+import net.opengis.wfs20.Wfs20Factory;
+import org.geoserver.wfs.Transaction.BatchManager;
+import org.geoserver.wfs.request.Delete;
+import org.geoserver.wfs.request.Insert;
+import org.geoserver.wfs.request.Native;
+import org.geoserver.wfs.request.Replace;
+import org.geoserver.wfs.request.TransactionElement;
+import org.geoserver.wfs.request.TransactionRequest;
+import org.geoserver.wfs.request.TransactionResponse;
+import org.geoserver.wfs.request.Update;
+import org.geotools.data.FeatureStore;
+import org.geotools.util.MapEntry;
+import org.junit.Test;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.filter.Filter;
+import org.opengis.filter.Or;
+
+/**
+ * Tests for {@link BatchManager}.
+ *
+ * @author awaterme
+ */
+public class BatchManagerTest {
```

Comment:
In GeoServer we don't have a separate group of testers that run consistent functional tests against the integrated product, everything has to be done by the developers. Can you please add a functional test for this as well, following the same pattern as the previous change, down in TransactionTest?

Reply:
Ok, I will add a further test case.

Reply:
Thank you.